src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageEndpoint.cs

```diff
@@ -156,7 +156,8 @@ public async Task<RazorMapToDocumentRangesResponse> Handle(RazorMapToDocumentRan
             await Task.Factory.StartNew(() =>
             {
                 _documentResolver.TryResolveDocument(request.RazorDocumentUri.GetAbsoluteOrUNCPath(), out documentSnapshot);
-                if (!_documentVersionCache.TryGetDocumentVersion(documentSnapshot, out documentVersion))
+                if (documentSnapshot is null ||
```

Comment:
If for whatever reason we couldn't get a documentSnapshot before, the following version check would throw a argument null exception.

Reply:
Not sure I follow. Okay with this change, or okay with allowing it to throw an argument null exception?

Reply:
Ok throwing an argument `null` exception. Something catastrophic happened and It'd be better to understand what imo

Reply:
> It'd be better to understand what imo

Totally agree, however would this not result in a silent failure as we return to the calling HTMLCSharpLanguageServer? If so, perhaps a `Debug.Fail` or `Log` would be a better bet?

Reply:
> Totally agree, however would this not result in a silent failure as we return to the calling HTMLCSharpLanguageServer? If so, perhaps a `Debug.Fail` or `Log` would be a better bet?

That'd be fine too